internal/gateway/handler.go

```diff
@@ -6,7 +6,11 @@ import (
 	"strings"
 )
 
-func ProvideHandler(config *Config, router *Router, auth JwtAuth) http.HandlerFunc {
+type Auth interface {
```

Comment:
you might be able to make this one private (other interfaces you moved to the consumer of them as well)